src/goLanguageServer.ts

```diff
@@ -79,7 +80,7 @@ export async function registerLanguageFeatures(ctx: vscode.ExtensionContext) {
 
 	// The user may not have the most up-to-date version of the language server.
 	const tool = getTool(toolName);
-	const update = await shouldUpdateLanguageServer(tool, path);
+	const update = await shouldUpdateLanguageServer(tool, path, config.checkForUpdates);
```

Comment:
Maybe just:

```typescript
if config.checkForUpdates {
    const update = await shouldUpdateLanguageServer(tool, path);
    if (update) {
        promptForUpdatingTool(toolName); 
    }
}
```

Reply:
I'd still want to check against the version that is hard-coded in the extension if the setting is disabled.
How about renaming the setting to `go.checkGoProxyForToolUpdates`?
For now, we can use it only for `gopls`. 
In the future the same approach can be taken for other tools.